databricks/koala/structures.py

```diff
@@ -156,6 +155,29 @@ def compute(self):
         """Alias of `toPandas()` to mimic dask for easily porting tests."""
         return self.toPandas()
 
+    @derived_from(pd.DataFrame)
+    def fillna(self, value=None, axis=None, inplace=False):
+        df = self.copy()
+        if axis is None:
+            axis = 0
+        if value is not None:
+            if axis == 0 or axis == "index":
+                if isinstance(value, (float, int, str, bool, dict)):
+                    df = df._spark_fillna(value)
+                elif isinstance(value, (pd.DataFrame, pd.Series)):
+                    raise NotImplementedError("Dataframe and Series value are not supported")
+            else:
+                raise NotImplementedError("fillna currently only works for axis=0 or axis='index'")
+        else:
+            raise ValueError('Must specify value')
+
+        if inplace:
+            df._metadata = self._metadata.copy()
+            _reassign_jdf(self, df)
```

Comment:
when `inplace=True`, what is the return? I would expect it to be `self`

Reply:
When `inplace=True`  it will return the result in the same dataframe. 

```
print(df.show())
+-----------------+----+----+----+
|__index_level_0__|   x|   y|   z|
+-----------------+----+----+----+
|               10|null| 1.0| 1.0|
|               20| 2.0| 2.0| 2.0|
|               30| 3.0|null| 3.0|
|               40| 4.0| 4.0| 4.0|
|               50|null|null|null|
|               60| 6.0|null|null|
+-----------------+----+----+----+

ddf.fillna({'x': -1, 'y': -2, 'z': -5}, inplace=True)
print(ddf.show())
+-----------------+----+----+----+
|__index_level_0__|   x|   y|   z|
+-----------------+----+----+----+
|               10|-1.0| 1.0| 1.0|
|               20| 2.0| 2.0| 2.0|
|               30| 3.0|-2.0| 3.0|
|               40| 4.0| 4.0| 4.0|
|               50|-1.0|-2.0|-5.0|
|               60| 6.0|-2.0|-5.0|
+-----------------+----+----+----+

```

Reply:
I changed it to 
```python
if inplace:
    self._jdf = df._jdf
```